stealth-addresses/src/cli.rs

```diff
@@ -0,0 +1,69 @@
+//! CLI definition and entrypoint to executable
+
+use clap::{Parser, Subcommand};
+use reth::args::{
+    utils::{chain_help, chain_value_parser, SUPPORTED_CHAINS},
+    LogArgs,
+};
+use reth_chainspec::ChainSpec;
+use reth_cli_commands::node::{self, NoArgs};
+use reth_cli_runner::CliRunner;
+use reth_db::DatabaseEnv;
+use reth_node_builder::{NodeBuilder, WithLaunchContext};
+use std::{future::Future, sync::Arc};
+
+/// Entrypoint.
+#[derive(Debug, Parser)]
+pub struct Cli {
+    /// The command to run
+    #[command(subcommand)]
+    command: Commands,
+
+    /// The chain this node is running.
+    ///
+    /// Possible values are either a built-in chain or the path to a chain specification file.
+    #[arg(
+        long,
+        value_name = "CHAIN_OR_PATH",
+        long_help = chain_help(),
+        default_value = SUPPORTED_CHAINS[0],
+        value_parser = chain_value_parser,
+        global = true,
+    )]
+    chain: Arc<ChainSpec>,
+
+    #[command(flatten)]
+    logs: LogArgs,
+}
+
+impl Cli {
+    pub fn run<L, Fut>(mut self, launcher: L) -> eyre::Result<()>
+    where
+        L: FnOnce(WithLaunchContext<NodeBuilder<Arc<DatabaseEnv>>>, NoArgs) -> Fut,
+        Fut: Future<Output = eyre::Result<()>>,
+    {
+        // add network name to logs dir
+        self.logs.log_file_directory =
+            self.logs.log_file_directory.join(self.chain.chain.to_string());
+
+        let _guard = self.logs.init_tracing()?;
```

Comment:
why do we need to do it separately here? `Node` command should already handle it for us I think